src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/NuGet/ProjectRestoreInfoBuilder.cs

```diff
@@ -0,0 +1,98 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using NuGet.SolutionRestoreManager;
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.IO;
+using System.Linq;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.NuGet
+{
+    internal static class ProjectRestoreInfoBuilder
+    {
+        private const string DefiningProjectDirectoryProperty = "DefiningProjectDirectory";
+        private const string ProjectFileFullPathProperty = "ProjectFileFullPath";
+
+        internal static IVsProjectRestoreInfo Build(IEnumerable<IProjectValueVersions> updates)
+        {
+            Requires.NotNull(updates, nameof(updates));
+
+            return Build(updates.Cast<IProjectVersionedValue<IProjectSubscriptionUpdate>>());
+        }
+
+        internal static IVsProjectRestoreInfo Build(IEnumerable<IProjectVersionedValue<IProjectSubscriptionUpdate>> updates)
+        {
+            Requires.NotNull(updates, nameof(updates));
+
+            // if none of the underlying subscriptions have any changes
+            if (!updates.Any(u => u.Value.ProjectChanges.Any(c => c.Value.Difference.AnyChanges)))
+            {
+                return null;
+            }
+
+            string baseIntermediatePath = null;
+            var targetFrameworks = new TargetFrameworks();
+            
+            foreach (IProjectVersionedValue<IProjectSubscriptionUpdate> update in updates)
+            {
+                var configurationChanges = update.Value.ProjectChanges[ConfigurationGeneral.SchemaName];
+                baseIntermediatePath = baseIntermediatePath ?? 
+                    configurationChanges.After.Properties[ConfigurationGeneral.BaseIntermediateOutputPathProperty];
+                string targetFrameworkMoniker = 
+                    configurationChanges.After.Properties[ConfigurationGeneral.TargetFrameworkMonikerProperty];
+                
+                if (!targetFrameworks.Contains(targetFrameworkMoniker))
+                {
+                    var projectReferencesChanges = update.Value.ProjectChanges[ProjectReference.SchemaName];
+                    var packageReferencesChanges = update.Value.ProjectChanges[PackageReference.SchemaName];
+
+                    targetFrameworks.Add(new TargetFrameworkInfo
+                    {
+                        TargetFrameworkMoniker = targetFrameworkMoniker,
+                        ProjectReferences = GetProjectReferences(projectReferencesChanges.After.Items),
+                        PackageReferences = GetReferences(packageReferencesChanges.After.Items)
+                    });
+                }
+            }
+
+            return new ProjectRestoreInfo
+            {
+                BaseIntermediatePath = baseIntermediatePath,
```

Comment:
/cc @alpaix may have more context on the scope of this first iteration. NuGet is able to generated full formed dg2 files with this information.

---
In reply to: [83518077](https://github.com/dotnet/roslyn-project-system/pull/524#discussion_r83518077) [](ancestors = 83518077)